spec/test_collector/test_links/reporter_spec.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'buildkite/test_collector/test_links_plugin/reporter'
+
+RSpec.describe Buildkite::TestCollector::TestLinksPlugin::Reporter do
+  let(:examples) { RSpec.world.filtered_examples }
+  let(:http_client) { double('Buildkite::TestCollector::HTTPClient') }
+  let(:suite_url) { 'https://example.com/suite/12345' }
+  let(:response) { { suite_url: suite_url }.to_json }
+  let(:io) { StringIO.new }
+  let(:reporter) { Buildkite::TestCollector::TestLinksPlugin::Reporter.new(io) }
+
+  before do
+    allow(Buildkite::TestCollector::HTTPClient).to receive(:new).and_return(http_client)
+    Buildkite::TestCollector.configure(
+      hook: :rspec,
+      token: 'fake_token',
+      url: 'http://fake.buildkite.localhost/v1/uploads'
+    )
+  end
+
+  after do
+    reset_io(io)
+  end
+
+  context 'when tests have failed' do
+    let!(:failed_example) do
+      OpenStruct.new(
+        example_group: OpenStruct.new(metadata: { full_description: 'I love to eat pies' }),
+        description: 'mince and cheese',
+        execution_result: OpenStruct.new(status: :failed)
+      )
+    end
+    let!(:notification) { RSpec::Core::Notifications::SummaryNotification.new(10.0, examples, [failed_example]) }
+
+    context 'there is no token' do
+      before do
+        Buildkite::TestCollector.configure(hook: :rspec, token: nil, url: 'http://fake.buildkite.localhost/v1/uploads')
+      end
+
+      it 'does not render summary' do
+        reporter.dump_failures(notification)
+        expect(io.string.strip).to be_empty
+      end
+
+      it 'does not request metadata' do
+        reporter.dump_failures(notification)
+        expect(Buildkite::TestCollector::HTTPClient).not_to receive(:new)
+      end
+    end
+
+    context 'fetch_metadata does not return a suite_url' do
+      it 'does not render summary' do
+        allow(http_client).to receive(:metadata).and_return(OpenStruct.new(code: '200',
```

Comment:
again just a 💅 thing: consider formatting like 

```
allow(http_client).to receive(:metadata)
                  .and_return(OpenStruct.new(code: '200', body: { suite_url: nil }.to_json))
```

(ok i cant get it to work super nicely in this view, but bascially to indent the `.to receive` and the `.and_return` at the same level)